lib/pagseguro/transaction_request/request_serializer.rb

```diff
@@ -143,6 +151,155 @@ def serialize_extra_params(extra_params)
+              xml.send(:number, transaction_request.billing_address.number)
+              xml.send(:complement, transaction_request.billing_address.complement)
+              xml.send(:district, transaction_request.billing_address.district)
+              xml.send(:city, transaction_request.billing_address.city)
+              xml.send(:state, transaction_request.billing_address.state)
+              xml.send(:country, transaction_request.billing_address.country)
+              xml.send(:postalCode, transaction_request.billing_address.postal_code)
+            end
+          end
+
+          if transaction_request.holder
+            xml.send(:holder) do
+              xml.send(:name, transaction_request.holder.name)
+              xml_serialize_documents(xml, [transaction_request.holder.document])
+
+              xml.send(:birthDate, transaction_request.holder.birth_date)
+              xml_serialize_phone(xml, transaction_request.holder.phone)
+            end
+          end
+        end
+      end
+
+      def xml_serialize_documents(xml, documents = [])
+        return if documents.reject(&:nil?).empty?
+
+        xml.send(:documents) {
+          documents.each do |document|
+            xml.send(:document) {
+              xml.send(:type, document.type)
+              xml.send(:value, document.value)
+            }
+          end
+        }
+      end
+
+      def xml_serialize_phone(xml, phone)
+        if phone
+          xml.send(:phone) {
+            xml.send(:areaCode, phone.area_code)
+            xml.send(:number, phone.number)
+          }
+        end
+      end
+
+      def xml_serialize_shipping(xml, shipping)
+        return unless shipping
+
+        xml.send(:shipping) do
+          if shipping.address
+            xml.send(:address) do
+              xml.send(:street, shipping.address.street)
+              xml.send(:number, shipping.address.number)
+              xml.send(:district, shipping.address.district)
+              xml.send(:country, shipping.address.country)
+              xml.send(:postalCode, shipping.address.postal_code)
+              xml.send(:city, shipping.address.city)
+              xml.send(:state, shipping.address.state)
+              xml.send(:complement, shipping.address.complement)
+            end
+          end
+        end
+      end
+
+      def xml_serialize_items(xml, items)
+        return unless items
+
+        xml.send(:items) do
+          items.each do |item|
+            xml.send(:item) do
+              xml.send(:id, item.id)
+              xml.send(:description, item.description)
+              xml.send(:quantity, item.quantity)
+              xml.send(:amount, to_amount(item.amount))
+            end
+          end
+        end
+      end
+
+      def xml_serialize_sender(xml, sender)
+        return unless sender
+
+        xml.send(:sender) {
+          xml.send(:name, sender.name)
+          xml.send(:email, sender.email)
+          xml_serialize_phone(xml, sender.phone)
```

Comment:
This methods (*name, email, phone...*) are required, right?
Because if they aren't present can broken the XML structure.